lib/crypto/src/crypto.erl

```diff
@@ -47,533 +56,572 @@
-                        | brainpoolP384r1
-                        | brainpoolP384t1
-                        | brainpoolP512r1
-                        | brainpoolP512t1
-                        | c2pnb163v1
-                        | c2pnb163v2
-                        | c2pnb163v3
-                        | c2pnb176v1
-                        | c2pnb208w1
-                        | c2pnb272w1
-                        | c2pnb304w1
-                        | c2pnb368w1
-                        | c2tnb191v1
-                        | c2tnb191v2
-                        | c2tnb191v3
-                        | c2tnb239v1
-                        | c2tnb239v2
-                        | c2tnb239v3
-                        | c2tnb359v1
-                        | c2tnb431r1
-                        | ipsec3
-                        | ipsec4
-                        | prime192v1
-                        | prime192v2
-                        | prime192v3
-                        | prime239v1
-                        | prime239v2
-                        | prime239v3
-                        | prime256v1
-                        | secp112r1
-                        | secp112r2
-                        | secp128r1
-                        | secp128r2
-                        | secp160k1
-                        | secp160r1
-                        | secp160r2
-                        | secp192k1
-                        | secp192r1
-                        | secp224k1
-                        | secp224r1
-                        | secp256k1
-                        | secp256r1
-                        | secp384r1
-                        | secp521r1
-                        | sect113r1
-                        | sect113r2
-                        | sect131r1
-                        | sect131r2
-                        | sect163k1
-                        | sect163r1
-                        | sect163r2
-                        | sect193r1
-                        | sect193r2
-                        | sect233k1
-                        | sect233r1
-                        | sect239k1
-                        | sect283k1
-                        | sect283r1
-                        | sect409k1
-                        | sect409r1
-                        | sect571k1
-                        | sect571r1
-                        | wtls1
-                        | wtls10
-                        | wtls11
-                        | wtls12
-                        | wtls3
-                        | wtls4
-                        | wtls5
-                        | wtls6
-                        | wtls7
-                        | wtls8
-                        | wtls9
-                          .
-
--type edwards_curve_dh() :: x25519 | x448 .
-
--type edwards_curve_ed() :: ed25519 | ed448 .
+-type ec_explicit_curve() ::
+    {Field :: ec_field(), Curve :: ec_curve(), BasePoint :: binary(), Order :: binary(),
+        % FIXME: Really?
+        CoFactor ::
```

Comment:
Is this on purpose? Or does it seem "weird" only because there's a comment? Also, the comment moved from "after `none`", to "before element"...

```erlang
{Field :: ec_field(),
 Curve :: ec_curve(),
 BasePoint :: binary(),
 Order :: binary(),
 CoFactor :: none % FIXME: Really?
           | binary()}.
```
^ how does this look? (I imply that `|` could be optioned to "start in next line".

Reply:
I have created an issue where we can start to discuss this design decision https://github.com/WhatsApp/erlfmt/issues/167

Reply:
> we have to pick one

I imagine that, like `elvis` (for example), there'll be configuration options (I'm sorry, I didn't read the whole manual). I do prefer operators before my code, in any case, albeit this is probably like discussing tabs v. spaces.

Reply:
No `erlfmt` is not going to provide many configuration options.
We have one which is the `width` at which we break expression up over lines, but that is it and we intend to keep this set very very minimal.  So lets continue the discussion :)

Reply:
FYI: We are doing some data analysis and early findings seems to suggest that we will change `erlfmt` and move pipes to the start of the line, instead of the end
https://github.com/WhatsApp/erlfmt/pull/186